aptos-move/framework/aptos-token/sources/token.move

```diff
@@ -520,11 +523,13 @@ module aptos_token::token {
             error::not_found(EBALANCE_NOT_PUBLISHED),
         );
         let balance = &mut table::borrow_mut(tokens, id).amount;
-        if (id.property_version == 0) {
+        if (id.property_version == 0 && *balance > amount) {
```

Comment:
Why add && *balance > amount?

Reply:
if balance == amount, I will delete the entry from the map. otherwise it is still there since the final balance is not 0,

Reply:
Can we remove the id.property_version == 0 check ? If there's a single version and it gets deleted, balance and amount should both be 1?

Reply:
Good point. @areshand

Reply:
another question is: is it possible that amount = 0, and return Token {amount: 0, ..}?

Reply:
@lightmark  this is a good catch. we should assert amount > 0. It doesn't make sense to withdraw 0 token.